static/js/src/advantage/subscribe/checkout/components/BuyButton/BuyButton.tsx

```diff
@@ -300,9 +300,16 @@ const BuyButton = ({ setError, products, action, coupon }: Props) => {
         localStorage.removeItem(key);
       });
 
-      const address = userInfo
-        ? `${userInfo?.customerInfo?.address?.line1} ${userInfo?.customerInfo?.address?.line2} ${userInfo?.customerInfo?.address?.city} ${userInfo?.customerInfo?.address?.postal_code} ${userInfo?.customerInfo?.address?.state} ${userInfo?.customerInfo?.address?.country}`
-        : `${values?.address} ${values?.postalCode} ${values?.city} ${values?.usState} ${values?.caProvince} ${values?.country}`;
+      const address = [
+        values?.address,
+        values?.postalCode,
+        values?.city,
+        values?.usState,
+        values?.caProvince,
+        values?.country,
+      ]
+        .filter(Boolean)
+        .join(" ");
```

Comment:
Down below, shouldn't you also change it to:
```
if (address) {
    formData.append("street", address);
}
```

That way you only set the field if it's present with a valid value.

Reply:
Thanks Allan, yep true! Updated 👍